enhancements/service-binding.md

```diff
@@ -0,0 +1,293 @@
+PostgreSQL database backing operator.
+``` yaml
+---
+[...]
+kind: CustomResourceDefinition
+apiVersion: apiextensions.k8s.io/v1beta1
+metadata:
+  name: databases.postgresql.baiju.dev
+  annotations:
+    servicebinding.dev/username: "path={.status.data.dbConfiguration},objectType=ConfigMap"
+    servicebinding.dev/password: "path={.status.data.dbConfiguration},objectType=ConfigMap"
+    service.binding/uri: "path={.status.data.url}"
+spec:
+  group: postgresql.baiju.dev
+  version: v1alpha1
+```
+
+### Operator Providing Metadata in OLM
+
+This guidance enables operator providers to specify binding information an
+operator's OLM (Operator Lifecycle Manager) descriptor. The Service Binding
+Operator extracts to bind the application together with the backing service.
+The information may be specified in the "status" and/or "spec" section of the
+OLM in plaintext or as a reference to a Secret or a ConfigMap.
+
+For example, this is a *bind-able* operator OLM Descriptor for a
+PostgreSQL database backing operator.
+``` yaml
+---
+[...]
+statusDescriptors:
+
+  description: Name of the Secret to hold the DB user and password
+    displayName: DB Password Credentials
+    path: dbCredentials
+    x-descriptors:
+      - urn:alm:descriptor:io.kubernetes:Secret
+      - servicebinding:user
+      - servicebinding:password
+
+  description: Database connection IP address
+    displayName: DB IP address
+    path: dbConnectionIP
+    x-descriptors:
+      - servicebinding
+```
+
+### User Providing Metadata in the CR or the Kubernetes resource
+
+```yaml
+---
+[...]
+
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  name: foo
+  namespace: bar
+  annotations:
+    service.binding/uri: "path={.spec.host}"
+spec:
+  host: example-sbose.apps.ci-ln-smyggvb-d5d6b.origin-ci-int-aws.dev.rhcloud.com
+  path: /
+  to:
+    kind: Service
+    name: example
+    weight: 100
+  port:
+    targetPort: 80
+  wildcardPolicy: None
+```
+
+A detailed guide to decorating backing services to make them binding friendly has been documented [here](https://github.com/application-stacks/service-binding-specification/blob/master/annotations.md).
+
+
+## Advanced Configuration
+
+### Custom Environment variables
+
+
+To make binding applications (e.g., legacy Java applications that depend on JDBC connectioon strings)  together with backing services more flexible, the Service Binding Operator supports the optional use of custom environment variables. To use custom environment variables, an application developer creates a ServiceBinding that looks like the one shown 
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+
+metadata:
+  name: binding-ecommerce
+  namespace: foo
+
+spec:
+
+  applications:
+  - resourceRef: payments-app
+    group: ""
+    version: v1
+    resource: deployments
+
+  services:
+  - group: postgresql.baiju.dev
+    version: v1alpha1
+    kind: Database
+    resourceRef: db-demo
+
+  customEnvVar:
```

Comment:
Should this be renamed to dataMapping as well? refer https://issues.redhat.com/browse/APPSVC-538

Reply:
done, thanks!